cfg/builder/builder_finalize.cc

```diff
@@ -258,22 +258,22 @@ void CFGBuilder::removeDeadAssigns(core::Context ctx, const CFG::ReadsAndWrites
 
             bool wasRead = RnW.readsSet[it->id].contains(bind.bind.variable.id()); // read in the same block
 
-            // TODO(jvilk): When I change the sort order of block args to be in LocalRef variable order, we can
-            // early-abort the loops below if `arg.variable > bind.bind.variable`.
             if (!wasRead) {
-                for (const auto &arg : it->bexit.thenb->args) {
-                    if (arg.variable == bind.bind.variable) {
-                        wasRead = true;
-                        break;
-                    }
+                // args are in LocalRef variable order
+                auto argIt = absl::c_lower_bound(it->bexit.thenb->args, bind.bind, [](auto &a, auto &b) -> bool {
```

Comment:
tbh, I almost left this as a linear search that early aborted. idk what the abstraction overhead of lower_bound is.